lib/memory_profiler/reporter.rb

```diff
@@ -100,12 +106,19 @@ def object_list(generation)
           class_name = helper.lookup_class_name(klass)
           gem        = helper.guess_gem(file)
 
-          string     = '' << obj  if klass == String
+          if klass == String
+            if mem_lite
+              string = helper.string_summary(ObjectSpace._id2ref(obj.__id__))
+              md5    = helper.lookup_string_digest(ObjectSpace._id2ref(obj.__id__))
+            else
+              string = '' << obj  if klass == String
+            end
+          end
 
           memsize = ObjectSpace.memsize_of(obj) + rvalue_size_adjustment
           # compensate for API bug
           memsize = rvalue_size if memsize > 100_000_000_000
-          result[obj.__id__] = MemoryProfiler::Stat.new(class_name, gem, file, location, memsize, string)
+          result[obj.__id__] = MemoryProfiler::Stat.new(class_name, gem, file, location, memsize, string, nil)
```

Comment:
The `md5` value needs to be passed here instead of `nil`

Reply:
Doh, sorry about this.

I was doing this all as a very blobby-no-tests based experiment, and then was trying to split things out into logical commits.  In doing so, I had this as `nil` in the first commit ("Add md5 to MemoryProfiler::Stat"), since the md5 calculation bit was not implemented yet, but would then be updated later... but clearly I forgot about that.

Also says I have some bad tests here, so need to make some additions to that.  Good find!